internal/service/transaction_service_test.go

```diff
@@ -216,6 +204,25 @@ func TestTransactionService_Withdraw(t *testing.T) {
 				owner:  owner,
 				amount: amount,
 			},
+			doMocks: func(deps transactionServiceDependencies) {
+				deps.accRepoMock.On("FindOne", ctx, owner).Return(models.Account{
+					AccountId: owner,
+					Name:      "Shankar",
+					LastName:  "Nakai",
+				}, nil)
+
+				deps.transRepoMock.On("GetBalance", ctx, owner).Return(models.Balance{
+					AccountId: owner,
+					Amount:    500,
+				}, nil)
+			},
+			wantErr: ErrInsufficentBalance,
+		},
+		"multi-transaction-consumption": {
```

Comment:
You should test 
- Multi-transaction consumption with exact value
- Multi-transaction consumption with Remaining value
- Multi-transaction consumption that errors out on the second marked transaction
   - The question you need to answer here is, do you think it should succeed on the previously marked transactions?